io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +196,189 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
 }
 
 impl NvmfSubsystem {
+    pub fn extract_nexus_name(nqn: &str) -> Option<String> {
+        let vec: Vec<String> = nqn.split(':').map(|s| s.to_string()).collect();
+
+        if vec.len() != 2 {
+            return None;
+        }
+        Some(vec[1].clone())
+    }
+
+    fn reset_cb(success: bool, ctx: *mut c_void) {
+        if success {
+            info!(
+                "nvmf_event_handler: controller successfully reset {:?}",
+                success
+            );
+        } else {
+            error!("nvmf_event_handler: failed to reset controller ");
+        }
+        done_cb(ctx, success);
+    }
+
+    pub async fn reset_one_child(dev_name: String, ctx: *mut c_void) {
```

Comment:
We could even omit this function and perform nexus child reset directly on device I/O handle which every nexus child has:

```
pub fn get_io_handle(
        &self,
    ) -> Result<Box<dyn BlockDeviceHandle>, CoreError> {
```

There is a 'reset()' function in this handle which hides low-level reset implementation

```
#[async_trait(?Send)]
pub trait BlockDeviceHandle {
   ...
       fn reset(&self, cb: IoCompletionCallback, cb_arg: IoCompletionCallbackArg) -> Result<(), CoreError>;
}
```

In case of NVMeOF replicas it performs exactly controller reset.

Reply:
looks like this reset is a type of IO from the initiator to reset a given channel, whereas we need to reset all the data channels. https://spdk.io/doc/group__bdev__io__submit__functions.html#gaca574d3e94be6150343823dd2e7ecf46